### PR TITLE
ADEN-7524

### PR DESCRIPTION
**Pull request by jacekbj**
Add parameters sent to queen_of_hearts model.

### CODE REVIEW

File: extensions/wikia/AdEngine/js/ml/billTheLizard.js [hunk 1]
```diff
@@ -65,7 +98,16 @@ define('ext.wikia.adEngine.ml.billTheLizard', [
 					top_1k: adContext.get('targeting.wikiIsTop1000') ? 1 : 0,
 					wiki_id: adContext.get('targeting.wikiId') || null,
 					video_id: featuredVideoData.mediaId || null,
-					video_tags: featuredVideoData.videoTags || null
+					video_tags: featuredVideoData.videoTags || null,
+					viewport_height: bucketizeViewportHeight(Math.max(
+						doc.documentElement.clientHeight, win.innerHeight || 0
+					)),
+					lang: pageParams.lang,
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/ml/billTheLizard.js]: Can we keep alphabetical order please? 😇

File: extensions/wikia/AdEngine/js/ml/billTheLizard.js [hunk 2]
```diff
@@ -114,6 +156,8 @@ define('ext.wikia.adEngine.ml.billTheLizard', [
 	return {
 		call: call,
 		hasResponse: hasResponse,
-		serialize: serialize
+		serialize: serialize,
+		// export for testing purpose
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/ml/billTheLizard.js]: Should we create a separate module for that then?

File: extensions/wikia/AdEngine/js/ml/billTheLizard.js [hunk 3]
```diff
@@ -45,11 +51,38 @@ define('ext.wikia.adEngine.ml.billTheLizard', [
...
 	function call() {
 		var config = instantGlobals.wgAdDriverBillTheLizardConfig || {},
 			featuredVideoData = adContext.get('targeting.featuredVideo') || {},
 			pageParams = pageLevelParams.getPageLevelParams();
 
+		var pv = pageParams.pv || 1;
+		pv = pv > 30 ? 30 : pv;
+
+		var pvGlobal = win.pvNumberGlobal || 1;
+		pvGlobal = pvGlobal > 40 ? 40 : pvGlobal;
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/ml/billTheLizard.js]: `Math.min(40, win.pvGlobal || 1);`?

File: extensions/wikia/AdEngine/js/ml/billTheLizard.js [hunk 4]
```diff
@@ -45,11 +51,38 @@ define('ext.wikia.adEngine.ml.billTheLizard', [
...
+		return bucket;
+	}
+
 	function call() {
 		var config = instantGlobals.wgAdDriverBillTheLizardConfig || {},
 			featuredVideoData = adContext.get('targeting.featuredVideo') || {},
 			pageParams = pageLevelParams.getPageLevelParams();
 
+		var pv = pageParams.pv || 1;
+		pv = pv > 30 ? 30 : pv;
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/ml/billTheLizard.js]: `Math.min(30, pageParams.pv || 1);`?

**Review by fraszczakszymon — APPROVED** [review]

**Review by dmnsobczak — APPROVED** [review]

**Pull request merged**